src/lib/rest/rest.cpp

```diff
@@ -616,6 +630,17 @@ static int contentTypeCheck(ConnectionInfo* ciP)
     return 1;
   }
 
+
+  // Case 5
+  if ((ciP->apiVersion == "v2") && (ciP->httpHeaders.contentType != "application/json"))
+  {
```

Comment:
As I commented before, it only makes the code more difficult to read (Fermin agrees with you on this however :-)).